src/sys/inotify.rs

```diff
@@ -125,18 +125,26 @@ bitflags! {
 /// An inotify instance. This is also a file descriptor, you can feed it to
 /// other interfaces consuming file descriptors, epoll for example.
 #[derive(Debug, Clone, Copy)]
-pub struct InotifyInstance {
+pub struct Inotify {
     fd: RawFd
 }
 
+/// This object is returned when you create a new watch on an inotify instance.
+/// It is then returned as part of an event once triggered. It allows you to
+/// know which watch triggered which event. 
+#[derive(Debug, Clone, Copy)]
```

Comment:
You should derive `Hash`, `PartialEq`, `Eq`, `PartialOrd`, and `Ord` for `WatchDescriptor`.  That way they can be stored in all the standard collections.